### ISSUE TITLE
carousel length not updating

### ISSUE THREAD

**Issue by josevalb** [Learus/react-material-ui-carousel#50]
I am sending different photos [] arrays to my carousel component which vary in data and length but only the initial length sets the carousel indicators after that it doesn't vary depending on the new length of the array. Please help!!

**Comment by Learus** [comment]
Hello. I'm currently trying to reproduce the problem. Stay tuned.

**Comment by Learus** [comment]
Ok, I figured out what was causing this bug. It is fixed on v1.9.0.

**Comment by josevalb** [comment]
Amazing is working now for resizing. However, it is not resetting to the first element in the array when I switch another element of my photos array. For example, if I navigate to the last element of array1 with length 6 and then switch to array2 with only length 2 it will be buggy. It should default to the first instance I think

**Comment by Learus** [comment]
To be fair, it should always update everything when the `children` change. That's a code design flaw I knew since the beginning. I will try to fix it though. Perhaps it demands a bigger change in the library than it seems at first glance.
Stay tuned as always.

**Comment by Learus** [comment]
I think it should have been fixed (v2.0.0). I made some serious changes to prop and state design and handling. Everything should be cleaner and make more sense code-wise now.
Thanks for pointing it out.

**Comment by josevalb** [comment]
Works fine. No more out of bounds. However, it seems to default to index 0 only when there is a length difference in the array pass to the component.

**Comment by Learus** [comment]
Yes, correct. I did that, as a safety measure, to not display empty space when the 2 arrays are of different length. 
I did not want check the 2 arrays for equality on each update. There are 2 reasons for that:
1. How do I define equality between 2 ReactNodes?
2. It would be too expensive and would hinder the carousel's performance. My goal is for it to be fast and simple most of all.

I hoped that a simple length check would be enough to combat unwanted behaviour. Perhaps, though, my solution does not suffice, in regards to consistent and expected functionality.

What's your opinion on it?
